src/сomponents/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  onChange: (el: number) => void,
+  onDelete: (el: number) => void,
+  edit: string,
+  onEdit: (el: string) => void,
+  onChangeName: (
+    el: React.KeyboardEvent<HTMLInputElement>,
+    currTodo: Todo,
+    currValue: string,
+  ) => void,
+};
+
+export const TodoList: React.FC<Props>
+  = ({
+    todos,
+    onChange,
+    onDelete,
+    edit,
+    onEdit,
+    onChangeName,
+  }) => {
+    const [currInputValue, setCurrInputValue] = useState(edit);
+
+    useEffect(() => {
+      localStorage.setItem('todos', JSON.stringify(todos));
+    }, [todos]);
+
+    return (
+      <ul className="todo-list" data-cy="todosList">
+        {todos.map(todo => (
+          <li
+            className={cn({
+              completed: todo.completed,
+              editing: todo.title === edit,
+            })}
+            key={todo.id}
+            onDoubleClick={() => {
+              onEdit(todo.title);
+              setCurrInputValue(todo.title);
+            }}
+          >
+            <div className="view">
+              <input
+                type="checkbox"
+                className="toggle"
+                id={`${todo.id}`}
+                checked={todo.completed}
+                onChange={() => onChange(todo.id)}
+              />
+              <label>
+                {todo.title}
+              </label>
+              <button
+                type="button"
+                aria-label="destroy"
+                className="destroy"
+                data-cy="deleteTodo"
+                onClick={() => onDelete(todo.id)}
```

Comment:
my suggestion to you is that it will be better to create the TodoItem component and move a big part of the logic from the app. tsx to that component.

It should improve the readability by at least